### ISSUE TITLE
stack installation is broken.

### ISSUE THREAD

**Issue by crocket** [travitch/taffybar#193]
I cloned this repository and used stack-8.0.yaml to build gtk2hs-buildtools and taffybar.
It failed.

```
    [1 of 2] Compiling Main             ( /tmp/stack1372/cairo-0.13.3.1/Setup.hs,
 /tmp/stack1372/cairo-0.13.3.1/.stack-work/dist/x86_64-linux/Cabal-1.24.2.0/setup
/Main.o )

    /tmp/stack1372/cairo-0.13.3.1/Setup.hs:5:1: error:
        Failed to load interface for ‘Gtk2HsSetup’
        Use -v to see a list of the files searched for.
```

`cabal install taffybar` worked miraculously without a glitch.

**Comment by sakshamsharma** [comment]
Did you install `gtk2hs-buildtools` before the installation / build of taffybar? As far as I remember, doing that had worked. #164 was the last PR on this thing, but is is incredibly outdated. I'd look into it but I'm running rather busy these days.

**Comment by crocket** [comment]
Let me try again.

**Comment by crocket** [comment]
I executed the following commands

```
stack --stack-yaml=stack-8.0.yaml install gtk2hs-buildtools
stack --stack-yaml=stack-8.0.yaml install
```

taffybar was installed into `~/.local/bin` by stack.

When I execute taffybar, I get the following errors.

```
$ taffybar -v
Configuration '/home/crocket/.config/taffybar/taffybar.hs' changed. Recompiling.
Error occurred while loading configuration file.
Launching custom binary /home/crocket/.cache/taffybar/taffybar-linux-x86_64

Error: 
/home/crocket/.config/taffybar/taffybar.hs:1:1: error:
    Failed to load interface for ‘System.Taffybar’
    Use -v to see a list of the files searched for.

/home/crocket/.config/taffybar/taffybar.hs:2:1: error:
    Failed to load interface for ‘System.Taffybar.Systray’
    Use -v to see a list of the files searched for.

/home/crocket/.config/taffybar/taffybar.hs:3:1: error:
    Failed to load interface for ‘System.Taffybar.TaffyPager’
    Use -v to see a list of the files searched for.

/home/crocket/.config/taffybar/taffybar.hs:4:1: error:
    Failed to load interface for ‘System.Taffybar.SimpleClock’
    Use -v to see a list of the files searched for.

/home/crocket/.config/taffybar/taffybar.hs:5:1: error:
    Failed to load interface for ‘System.Taffybar.Text.CPUMonitor’
    Use -v to see a list of the files searched for.
```

~/.config/taffybar/taffybar.hs

```
import System.Taffybar
import System.Taffybar.Systray
import System.Taffybar.TaffyPager
import System.Taffybar.SimpleClock
import System.Taffybar.Text.CPUMonitor

main :: IO ()
main = do
  let pager = taffyPagerNew defaultPagerConfig
      tray = systrayNew
      cpu = textCpuMonitorNew "CPU : $total$%" 1
  defaultTaffybar defaultTaffybarConfig { startWidgets = [ pager ]
                                        , endWidgets = [ tray, cpu ]
                                        }

```

~/.cabal/bin/taffybar succeeds in compilation, but ~/.local/bin/taffybar does not.

There's something wrong with my stack installation, and I don't know what went wrong, yet.

**Comment by travitch** [comment]
Adding gtk2hs-buildtools to the build-depends of taffybar seems to cause very annoying build failures under Travis, so I had to remove it.  As for stack, those stack files are mostly for building taffybar under travis to validate commits.  You won't be able to run the resulting executable because the dyre library doesn't have stack support yet (at least, last time I checked).

**Comment by crocket** [comment]
I think you need to mention why stack build currently doesn't work on `README.md`.

**Comment by sakshamsharma** [comment]
@crocket Please try to run `taffybar` the first time using `stack exec taffybar -v`. I remember that worked for me. It is only needed the first time you run taffybar after changes to the configuration. If that does not work, run the same command from inside your taffybar source.

**Comment by crocket** [comment]
Executing `stack exec --stack-yaml=stack-8.0.yaml taffybar -v` in the repository folder works, but `~/.local/bin/taffybar` still doesn't work.

**Comment by shwsun** [comment]
stack install is breaking now, I think using cabal or even clone the repo local and them compile things from there work fine

**Comment by IvanMalison** [comment]
> You won't be able to run the resulting executable because the dyre library doesn't have stack support yet (at least, last time I checked).

@travitch Actually, dyre supports stack to the extent that it is needed for our purposes in git, but they haven't yet posted this version to hackage/stackage. I've actually been able to get taffybar running completely in stack with this stack.yaml

https://github.com/IvanMalison/dotfiles/blob/be630b620c9f8299053d920dfbacb3f18303bb91/dotfiles/config/taffybar/stack.yaml#L9

I think we should consider using the git master version of dyre so that we can support this for now.

**Comment by IvanMalison** [comment]
#218 should allow proper stack builds. It includes a quick-start script that should help anyone who is interested to get things running.

**Comment by IvanMalison** [comment]
The solution here, as @sakshamsharma suggested here https://github.com/travitch/taffybar/issues/194#issuecomment-296206263

is for gtk2hs to add these as build dependencies.

Not really an issue with taffybar, but I'll leave this open until this is fixed (or #256 is done)

**Comment by IvanMalison** [comment]
@sakshamsharma I can't tell, can we fix this now with the changes that have occured in gtk2hs?

**Comment by Tordek** [comment]
Is this fixed with #256 complete?
